internal/security/fileprovider/provider.go

```diff
@@ -77,14 +77,25 @@ func (p *fileTokenProvider) Run() error {
 		return err
 	}
 
+	tokenConfEnv, err := GetTokenConfigFromEnv()
+	if err != nil {
+		p.logger.Error(fmt.Sprintf("failed to get token config from environment variable %s with error: %s", addSecretstoreTokensEnvKey, err.Error()))
+		return err
+	}
+
 	var tokenConf TokenConfFile
 	if err := LoadTokenConfig(p.fileOpener, p.tokenConfig.ConfigFile, &tokenConf); err != nil {
 		p.logger.Error(fmt.Sprintf("failed to read token configuration file %s: %s", p.tokenConfig.ConfigFile, err.Error()))
 		return err
 	}
 
-	for serviceName, serviceConfig := range tokenConf {
+	// merge the additional token configuration list from environment variable
+	// note that the configuration file takes precedence, as the tokenConf will override
+	// the tokenConfEnv with same duplicate keys
+	// The tokenConfEnv only uses default settings.
+	tokenConf = tokenConfEnv.mergeWith(tokenConf)
```

Comment:
@jim-wang-intel I thought the point of @lenny-intel's comment was to do
tokenConf = tokenConf.mergeWith(tokenConfEnv)
where the first act of business is to determine whether tokenConfEnv is empty and return promptly out of merge. This way you could even just return an None/nil object instead of instantiating an empty map.

Reply:
@mkbhanda , nope it was to and the comment so it was clear that the current call is doing as we expected.

Reply:
I added comments in the code above already- let me know if this is clear.

Reply:
@jim-wang-intel , looks good.